workflows/reusable_integration_test.yml

```
@@ -0,0 +1,59 @@
+name: "Integration Tests"
+
+on:
+    workflow_call:
+      inputs:
+        repository:
+          required: true
+          type: string
+        branch:
+            required: true
+            type: string
+        environment:
+            required: true
+            type: string
+
+jobs:
+  IntegrationTests:
+    name: Integration Tests
+    runs-on: ubuntu-latest
+    environment: ${{inputs.environment}}
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v4
+        if: ${{inputs.branch == 'mainline'}}
+        with:
+          ref: mainline
+      
+      - uses: actions/checkout@v4
+        if: ${{inputs.branch == 'release'}}
+        with:
+          ref: release
```

Can this be simplified to this?
```suggestion
      - uses: actions/checkout@v4
        if: ${{inputs.branch == 'mainline' || inputs.branch == 'release'}}
        with:
          ref: ${{inputs.branch}}
```

Same thing for `workflows/reusable_canary.yml`

---

Yeah good call. I'll fix that.